### PR TITLE
env vars are strings not arrays

### LINKED ISSUE #1868: NoMethodError: undefined method `split' for [""]:Array
# Issue summary

Hello, I am new to shopify apps and am having trouble serving my app. I have installed ngrok and ran shopify app tunnel auth <token>
I then serve my app which produces a unexpected error without any message to what went wrong. Can anyone help

`X An unexpected error occured.
        To ←]8;;https://github.com/Shopify/shopify-cli/issues/new?title=NoMethodError%3A+undefined+method+%60split%27+for+%5B%22%22%5D%3AArray&body=%23+Issue+summary%0A%0AWrite+a+short
+description+of+the+issue+here+%E2%86%93%0A%0A%0A%0A%23%23+Expected+behavior%0A%0AWhat+do+you+think+should+happen%3F%0A%0A%0A%0A%23%23+Actual+behavior%0A%0AWhat+actually+happens%3F%0A%
0ATip%3A+include+an+error+message+%28in+a+%60%3Cdetails%3E%3C%2Fdetails%3E%60+tag%29+if+your+issue+is+related+to+an+error%0A%0A%0A%0A%23%23+Steps+to+reproduce+the+problem%0A%0A1.%0A1.%
0A1.%0A%0A%0A%23%23+Stacktrace%0A%0AC%3A%2FRuby27-x64%2Flib%2Fruby%2Fgems%2F2.7.0%2Fgems%2Fshopify-cli-2.7.3%2Fvendor%2Fdeps%2Fcli-kit%2Flib%2Fcli%2Fkit%2Fsystem.rb%3A225%3Ain+%60which
%27%0AC%3A%2FRuby27-x64%2Flib%2Fruby%2Fgems%2F2.7.0%2Fgems%2Fshopify-cli-2.7.3%2Fvendor%2Fdeps%2Fcli-kit%2Flib%2Fcli%2Fkit%2Fsystem.rb%3A219%3Ain+%60resolve_path%27%0AC%3A%2FRuby27-x64
%2Flib%2Fruby%2Fgems%2F2.7.0%2Fgems%2Fshopify-cli-2.7.3%2Fvendor%2Fdeps%2Fcli-kit%2Flib%2Fcli%2Fkit%2Fsystem.rb%3A126%3Ain+%60system%27%0AC%3A%2FRuby27-x64%2Flib%2Fruby%2Fgems%2F2.7.0%
2Fgems%2Fshopify-cli-2.7.3%2Flib%2Fshopify_cli%2Fcontext.rb%3A438%3Ain+%60system%27%0AC%3A%2FRuby27-x64%2Flib%2Fruby%2Fgems%2F2.7.0%2Fgems%2Fshopify-cli-2.7.3%2Flib%2Fshopify_cli%2Fser
vices%2Fapp%2Fserve%2Fphp_service.rb%3A39%3Ain+%60block+in+call%27%0A%0A%0A%23%23+Specifications%0A%0A-+App+type%3A%0A-+Operating+System%3A%0A-+Shell%3A%0A-+Ruby+version+%28ruby+-v%29%
3A%0A&labels=type%3Abug←\submit an issue←]8;;←\ `

### PR DESCRIPTION
**Pull request by amcaplan**
We were trying to `split` the resulting array, which failed!

This comes up in PHP apps because the subprocess doesn't have PATHEXT set.

### WHY are these changes introduced?

Fixes #1868 #1858 #1828 #1783 #1704 #1569 (perhaps more but those are the ones I found)

Apparently when we don't find a `$PATHEXT` env variable, we were substituting with an array instead of a string.

### WHAT is this pull request doing?

We try to use `$PATHEXT` but if it's not present, just use `['']`.

Basically we were trying to do that before, but we now do the `split` only if we found a value.

### How to test your changes?

On Windows:

```
shopify app create php
shopify app serve
```

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).

### CODE REVIEW

File: vendor/deps/cli-kit/lib/cli/kit/system.rb [hunk 1]
```diff
@@ -227,7 +227,7 @@ def resolve_path(a, env)
         end
 
         def which(cmd, env)
-          exts = os == :windows ? env.fetch('PATHEXT', ['']).split(';') : ['']
+          exts = (os == :windows && env['PATHEXT']&.split(';')) || ['']
```
Comment (amcaplan) [inline vendor/deps/cli-kit/lib/cli/kit/system.rb:230]: Alternatively:

```suggestion
          exts = 
            if os == :windows
              env['PATHEXT']&.split(';') || ['']
            else
              ['']
            end
```

Thoughts?

**Comment by amcaplan** [comment]
Good point about the title. I forget it's going to appear in the changelog!

Looks like [CLI-kit](https://github.com/Shopify/cli-kit/blob/823b02426e07f09d2a9ec675ec426c98d3e95535/lib/cli/kit/system.rb#L334) is a few versions behind. Do we usually push changes upstream?
